### PR TITLE
Fix 4816

### PR DESCRIPTION
**Pull request by matt852**
# Closes: #4816
# What's Changed

Lots of bug fixes for Dark Mode!

Fixed navbar link colors in dark mode.
Fixed hover over effect for navbar links in dark mode (now shows up as orange, just like in light mode).
Fixed spacing between buttons in Table Configuration modal.
Fixed background color for table rows (ie: table listing all devices).
Fixed nav tabs borders.
Fixed selected text not showing as selected on iGraphQL page.
Fixed black background in Swagger API docs for each row (when expanded).
Updated theme selection modal screenshots.
Slightly updated text color to make it easier to read.
Added missing inline comments to dark mode CSS file.

# TODO

- [X] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/project-static/css/dark.css [hunk 1]
```diff
@@ -63,10 +63,14 @@ html[data-theme="dark"] .panel-heading,                   /* Panel header bars *
 html[data-theme="dark"] .panel-footer,                    /* Panel footer bars */
 html[data-theme="dark"] .modal-header,                    /* Header bar in modals */
 html[data-theme="dark"] .modal-footer,                    /* Footer bar in modals */
-html[data-theme="dark"] .table-headings th {
+html[data-theme="dark"] .table-headings th {              /* Table header bar */
   background-color: rgb(196, 204, 210) !important;
 }
 
+html[data-theme="dark"] .table-headings tr {              /* Table rows */
+  background-color: rgb(217, 217, 217) !important;
```
[the remark this answers is not in the document]
Comment (matt852) [reply nautobot/project-static/css/dark.css]: Good call!  I updated the background to match the table rows on the homepage.
Screenshot 2023-11-16 at 12 29 06 PM

**Review by HanlinMiao — COMMENTED** [review]
The only thing I noticed is that the outline on the tab content/panel is not noticeable anymore in dark mode as opposed to in light mode. Wonder if this is something we want to address in this PR as well.
Screen Shot 2023-11-16 at 3 29 38 PM
Screen Shot 2023-11-16 at 3 29 48 PM

**Comment by matt852** [comment]
@HanlinMiao Good catch, I didn't realize this either.  Updated!

**Review by HanlinMiao — APPROVED** [review]
Looks great tysm!

**Review by timizuoebideri1 — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

File: nautobot/dcim/templates/dcim/location.html [hunk 2]
```diff
@@ -173,7 +173,7 @@
         <div class="panel-heading">
             <strong>Stats</strong>
         </div>
-        <div class="row panel-body">
+        <div class="panel-body">
```
Comment (HanlinMiao) [inline nautobot/dcim/templates/dcim/location.html:176]: This was causing the panel to extend outwards of the panel boundary it is now fixed.
Screen Shot 2023-11-20 at 12 33 14 PM

Comment (HanlinMiao) [reply nautobot/dcim/templates/dcim/location.html:176]: Screen Shot 2023-11-20 at 12 34 57 PM

**Pull request merged**